packages/next/src/server-build.ts

```diff
@@ -1090,6 +1089,36 @@ export async function serverBuild({
     )
   );
 
+  let pagesPlaceholderRscEntries;
+
+  if (appDir) {
+    // since we attempt to rewrite all paths to an .rsc variant,
+    // we need to create dummy rsc outputs for all pages entries
+    // this is so that an RSC request to a `pages` entry will match
+    // rather than falling back to a catchall `app` entry
+    // on the nextjs side, invalid RSC response payloads will correctly trigger an mpa navigation
+    const pagesManifest = path.join(
+      entryPath,
+      outputDirectory,
+      `server/pages-manifest.json`
+    );
+
+    const pagesData = await fs.readJSON(pagesManifest);
+    const pagesEntries = Object.keys(pagesData);
+
+    if (pagesEntries.length > 0) {
+      const dummyFilePath = path.join('./', 'dummyFile');
+      await fs.writeFile(dummyFilePath, 'RSC Placeholder');
+
+      pagesPlaceholderRscEntries = pagesEntries.reduce((acc, page) => {
+        acc[`${page.slice(1)}.rsc`] = new FileFsRef({
+          fsPath: dummyFilePath,
```

Comment:
I assumed this would fail since I thought `fsPath` needed to be an absolute path and this looks like a relative path